src/core/PasswordGenerator.cpp

```diff
@@ -220,6 +285,29 @@ QVector<PasswordGroup> PasswordGenerator::passwordGroups() const
         passwordGroups.append(group);
     }
 
+    int i = 0;
+    while (1) {
```

Comment:
It's not very clear at first look what this piece of code is suppose to do, insert a comment or move it into a function

Reply:
How about that :"Loop over character groups and remove excluded characters from them; remove empty groups"? Also break condition should be moved from `if` to `while`

Reply:
Also, you should use while(true) instead. You are implicitly converting 1 to bool in this case.

Reply:
@seregaxvm I think that can do it. :wink:

Reply:
done